fep/escseq.c

```diff
@@ -941,6 +955,36 @@ static void set_attr(const char *str, int len)
             s_attr.foreground = params[i];
           } else if ((40 <= params[i] && params[i] <= 47) || (100 <= params[i] && params[i] <= 107)) {
             s_attr.background = params[i];
+          } else if (params[i] == 38 || params[i] == 48) {
+            int j;
+            int *ext_color_dst = params[i] == 38 ? &s_attr.foreground : &s_attr.background;
+            i++;
+            if (i < nr_params) {
+              if (params[i] == 2 || params[i] == 5) {
+                int ext_param_num;
+                if (params[i] == 5) {
+                  *ext_color_dst = EXT_COLOR_256;
+                  ext_param_num = 1;
+                }
+                else
+                {
+                  *ext_color_dst = EXT_COLOR_24BIT;
+                  ext_param_num = 3;
+                }
+
+                for(j = 0; j < ext_param_num; j++) {
+                  i++;
+                  if (i < nr_params) {
```

Comment:
How about moving this condition to `for`?

```c
for (j = 0; j < ext_param_num && i < nr_params; j++, i++) {
```

Reply:
Thanks for the suggestion.
I can include the condition `i < nr_params` in the `for` as shown below. However, if I include `i++` in the `for`, it will be double incremented in the outer for.
```c
for (j = 0; j < ext_param_num && i < nr_params; j++) {
  i++;
```